tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/ExecutionEnvironment.java

```diff
@@ -41,6 +44,15 @@
    */
   Types environmentType();
 
+  Map<String, MutableVariable<?>> variables();
+
+  /**
+   * Registers a variable with this execution environment.
+   * @deprecated Done automatically in Variable's constructor, should only be used internally.
+   */
+  @Deprecated
+  void registerVariable(MutableVariable<?> variable);
```

Comment:
Yeah, that's correct, I used `@Deprecated` as I'm not aware of a more semantically correct version, and just noting it in the doc seemed insufficient.  Is there a more correct strategy for this?

Reply:
Refactor it so that it's not visible, or document that it's not for external use (though that's less than ideal). The deprecated tag will probably cause the linter to fail on TF-java builds when we re-enable it, so we can't do that.

Reply:
Unfortunately, it's in an interface, so javadoc is the best I can do.  Manual use shouldn't cause too many issues though, since it's keyed on a final field.